google/cloud/aiplatform/metadata/metadata.py

```diff
@@ -46,12 +49,30 @@ def set_run(self, run: str):
                 "No experiment set for this run. Make sure to call aiplatform.init(experiment='my-experiment') "
                 "before trying to set_run. "
             )
+        run_execution_id = f"{self._experiment}-{run}"
         execution = _Execution.get_or_create(
-            resource_id=run,
+            resource_id=run_execution_id,
+            display_name=run,
             schema_title=constants.SYSTEM_RUN,
             schema_version=constants.SCHEMA_VERSIONS[constants.SYSTEM_RUN],
         )
+        _Context.add_artifacts_or_executions(
+            context_id=self._experiment, execution_ids=[execution.name]
+        )
+
+        metrics_artifact_id = f"{self._experiment}-{run}-metrics"
+        artifact = _Artifact.get_or_create(
+            resource_id=metrics_artifact_id,
+            display_name=metrics_artifact_id,
+            schema_title=constants.SYSTEM_METRICS,
+            schema_version=constants.SCHEMA_VERSIONS[constants.SYSTEM_METRICS],
+        )
+        _Execution.add_artifact(
+            execution_id=run_execution_id, artifact_id=metrics_artifact_id, input=False
+        )
+
         self._run = execution.name
+        self._metric = artifact.name
```

Comment:
Same comment about setting the _run and _metric as the underlying object.

Reply:
I think for run and metrics, we might prefer to just record the resource ID here, the reason is that those resources would only be used in log_xxx, and get_xxx calls later. And we would want to query our BE to get the most up to date resources for logging data instead of using the one that was queried at set_run(), which might be out of date. 

I agree that the object would also include the resource ID/name, but I think it might cause some confusion about why we would not use them directly in the log_xxx calls and instead query our BE API for a new instances.

Let me know what you think.

Reply:
Changed my mind based you and Nacho's comments, now all 3 are storing resources.